### PR TITLE
[LLM] - [LIVE-9909] - Add support for relatives import in `tsconfig.json`

### PR DESCRIPTION
**Pull request by sshmaxime**
### 📝 Description

This PR aims at adding supports for path mapping on LLM's bundler (Metro).

**Why:** It very cumbersome to see lots of `../../../` and such imports in our source code. This PR aims at solving this issue. It shall allow us to organize our code in better ways along the fact of making it more readable.

**Why2:** `tsconfig.json` paths mappings are resolved in VSCode or your IDE but the bundler doesn't know about them and therefore that's why we cannot use them as of now. There is a package [babel-plugin-module-resolver](https://www.npmjs.com/package/babel-plugin-module-resolver) that's supposed to do that for us but after a lot of failed attempts on trying to get it working (which maybe has to do with the custom module resolver we added to Metro - but maybe not, I found a lot of issues or posts about people saying that this package doesn't work well with `react-native`) I took the initiative on doing that internally.

**How to use it:** 

Add any desired path mapping in `tsconfig.json` on LLM: (for instance `"@utils/*": ["./src/utils/*"]`)
Then, import `@utils/constants` in your project files and it will automatically resolve to `./src/utils/constants.{js/jsx/ts/tsx}`.

Please respect the following structure: `"@{package}/*": ["{any/path}/*"]`
```
// tsconfig.json
 {
    { ... }
    "paths": {
      "@utils/*": ["./src/utils/*"],
      "@constants/*": ["./src/constants/*"],
    }
  }
```

### ❓ Context

- **Impacted projects**: `LLM` 
- **Linked resource(s)**: [LIVE-9909] 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-9909]: https://ledgerhq.atlassian.net/browse/LIVE-9909?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tools/metro-extra-config/index.js [hunk 1]
```diff
@@ -8,15 +8,14 @@
 /* eslint-disable no-console */
 
 const path = require("path");
-const defaultSourceExts =
-  require("metro-config/src/defaults/defaults").sourceExts;
+const defaultSourceExts = require("metro-config/src/defaults/defaults").sourceExts;
 const { makeMetroConfig } = require("@rnx-kit/metro-config");
 const MetroSymlinksResolver = require("@rnx-kit/metro-resolver-symlinks");
-const resolve = require("metro-resolver").resolve;
```
Comment (sshmaxime) [inline tools/metro-extra-config/index.js:15]: It wasn't used.

File: tools/metro-extra-config/index.js [hunk 2]
```diff
@@ -116,11 +105,18 @@ module.exports = function (options = {}, config = {}) {
...
-            // ignore for now…
+          } catch (e) {
+            console.log(e);
+          }
+
+          try {
+            const resolvedModule = tsConfigResolver(moduleName);
+            if (resolvedModule) return resolvedModule;
+          } catch (e) {
+            console.log(e);
```
Comment (sshmaxime) [inline tools/metro-extra-config/index.js:119]: I think it's better to log the errors than just ignoring them.

File: tools/metro-extra-config/package.json [hunk 3]
```diff
@@ -7,6 +7,7 @@
   "dependencies": {
     "@rnx-kit/metro-config": "^1.3.2",
     "@rnx-kit/metro-resolver-symlinks": "0.1.32",
+    "json5": "^2.2.3",
```
Comment (sshmaxime) [inline tools/metro-extra-config/package.json:10]: Needed to read the `tsconfig.json` file with comments.

File: apps/ledger-live-mobile/tsconfig.json [hunk 4]
```diff
@@ -8,14 +8,14 @@
     "types": ["jest", "@testing-library/jest-native/extend-expect"],
     "noEmit": true,
     "target": "esnext",
-    "incremental": true
+    "incremental": true,
+    "moduleResolution": "node",
+    "baseUrl": ".",
+    "paths": {
```
Comment (valpinkman) [inline apps/ledger-live-mobile/tsconfig.json:14]: nice

File: tools/metro-extra-config/package.json (hunk 3 above)
Comment (valpinkman) [reply tools/metro-extra-config/package.json:10]: do we really want comments on the json file ? it's not standard

**Review by valpinkman — APPROVED** [review]

File: tools/metro-extra-config/package.json (hunk 3 above)
Comment (sshmaxime) [reply tools/metro-extra-config/package.json]: removed

**Pull request merged**